client/lib/gsuite/index.js

```diff
@@ -107,10 +107,11 @@ function getGSuiteSupportedDomains( domains ) {
 			return false;
 		}
 
-		const isHostedOnWpcom = isRegisteredDomain( domain ) && ( domain.hasWpcomNameservers || hasGSuiteWithUs( domain ) );
-		const isMapped = isMappedDomain( domain );
+		const isHostedOnWpcom =
+			isRegisteredDomain( domain ) && ( domain.hasWpcomNameservers || hasGSuiteWithUs( domain ) );
+		const isMappedWithWpcomNameservers = isMappedDomainWithWpcomNameservers( domain );
 
-		return ( isHostedOnWpcom || isMapped ) && canDomainAddGSuite( domain.name );
+		return ( isHostedOnWpcom || isMappedWithWpcomNameservers ) && canDomainAddGSuite( domain.name );
```

Comment:
We could avoid the unnecessary `isMappedWithWpcomNameservers` variable by structure our code differently:

```javascript
if ( ! isHostedOnWpcom && ! isMappedDomainWithWpcomNameservers( domain ) ) {
  return false;
}

return canDomainAddGSuite( domain.name );
```